accounts/abi/abi.go

```diff
@@ -274,12 +274,21 @@ func (abi ABI) Unpack(v interface{}, name string, output []byte) error {
 		// struct will match named return values to the struct's field
 		// names
 		case reflect.Struct:
+			idx := 0
 			for i := 0; i < len(method.Outputs); i++ {
-				marshalledValue, err := toGoType(i, method.Outputs[i], output)
+				marshalledValue, err := toGoType(idx, method.Outputs[i], output)
 				if err != nil {
 					return err
 				}
 				reflectValue := reflect.ValueOf(marshalledValue)
+				kind := reflectValue.Type().Kind()
```

Comment:
This could be `typ := reflectValue.Type()` and you won't have to call `Type` multiple times here, and in the `if` clause.